hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/SlowPeerDisabledTracker.java

```diff
@@ -46,11 +46,6 @@ public class SlowPeerDisabledTracker extends SlowPeerTracker {
 
   public SlowPeerDisabledTracker(Configuration conf, Timer timer) {
     super(conf, timer);
-    final boolean dataNodePeerStatsEnabledVal =
-        conf.getBoolean(DFSConfigKeys.DFS_DATANODE_PEER_STATS_ENABLED_KEY,
-            DFSConfigKeys.DFS_DATANODE_PEER_STATS_ENABLED_DEFAULT);
-    Preconditions.checkArgument(!dataNodePeerStatsEnabledVal,
-        "SlowPeerDisabledTracker should only be used for disabled slow peer stats.");
```

Comment:
The conf obtained here is the one currently running in the system, so delete these verification logic.

NameNode#reconfigureSlowNodesParameters => datanodeManager.initSlowPeerTracker(getConf(), timer, peerStatsEnabled) => getConf() => dataNodePeerStatsEnabledVal==true

Reply:
Maybe we need to keep this verification logic.
when reconfigureSlowNodesParameters we can set the DFS_DATANODE_PEER_STATS_ENABLED_KEY value in the conf, such as:
```
case DFS_DATANODE_PEER_STATS_ENABLED_KEY: {
         Timer timer = new Timer();
         if (newVal != null && !newVal.equalsIgnoreCase("true") && !newVal.equalsIgnoreCase(
             "false")) {
           throw new IllegalArgumentException(newVal + " is not boolean value");
         }
         final boolean peerStatsEnabled = newVal == null ?
             DFS_DATANODE_PEER_STATS_ENABLED_DEFAULT :
             Boolean.parseBoolean(newVal);
         result = Boolean.toString(peerStatsEnabled);
         getConf().setBoolean(DFS_DATANODE_PEER_STATS_ENABLED_KEY, peerStatsEnabled);
         datanodeManager.initSlowPeerTracker(getConf(), timer, peerStatsEnabled);
         break;
       }
```

Reply:
1. The logic here may not be correct because the system is now changing the value of the key from true to false, while the key of getConf is the old key, throw IllegalArgumentException in Preconditions.checkArgument.
<img width="1100" alt="image" src="https://github.com/apache/hadoop/assets/63718681/613e610f-3999-4960-9adb-0b9fed18dcdf">
---<img width="944" alt="image" src="https://github.com/apache/hadoop/assets/63718681/7495c5e2-c8d4-479f-9f3b-4f4eda5d7787">


2. It does not affect reconfig DFS_ DATANODE_ PEER_ STATS_ ENABLED_ KEY's logic

3. TestNameNodeReconfigure#testSlowPeerTrackerEnabled UT skipped the logic of refreshing the configuration, so when true is changed to false, the original value in the system remains false. Therefore, the following code passed the UT
<img width="1181" alt="image" src="https://github.com/apache/hadoop/assets/63718681/8772ae17-6da0-49e9-8b83-de7da84b63e2">

4. When to refresh the configuration, refer to the following code(ReconfigurableBase#reconfigureProperty)
<img width="941" alt="image" src="https://github.com/apache/hadoop/assets/63718681/e3dd7d4e-1649-472f-8d68-2f0fc436a8e3">

Reply:
Yes, the logic of the testSlowPeerTrackerEnable UT called nameNode.reconfigurePropertyImpl and it will not refresh the system runtime configuration, Currently, it does not affect this UT.

I think we can create a new PR to fix this issue. That is to change nameNode.reconfigurePropertyImpl to nameNode.reconfigureProperty, and add some other cases.

Reply:
Sorry, I just understood what you mean. This is very nice.

In addition, I will replace the nameNode.reconfigurePropertyImpl used in testSlowPeerTrackerEnable UT with nameNode.reconfigurePropertyImpl, which is more in line with the production env.

Reply:
> here add logic getConf().setBoolean(DFS_DATANODE_PEER_STATS_ENABLED_KEY, peerStatsEnabled);

I think this should be done in separate PR for all configs, just the test change is enough.

Reply:
@haiyang1987 
we will proceed with the original processing, as there is no validation in SlowPeerTracker. 
In SlowPeerDisabledTracker, we will first remove this invalid validation. ‘getConf().setBoolean(DFS_DATANODE_PEER_STATS_ENABLED_KEY, peerStatsEnabled);‘.

If ‘setConf’ in advance, the logic of the original reconfig may have a slightly different meaning, because in the production environment, the configuration is changed through reconfigureProperty and then the configuration in memory is refreshed.